drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -1529,8 +1529,11 @@ void RasterizerSceneGLES3::_setup_environment(const RenderDataGLES3 *p_render_da
 		}
 
 		scene_state.ubo.fog_enabled = environment_get_fog_enabled(p_render_data->environment);
+		scene_state.ubo.fog_mode = environment_get_fog_mode(p_render_data->environment);
 		scene_state.ubo.fog_density = environment_get_fog_density(p_render_data->environment);
-		scene_state.ubo.fog_height = environment_get_fog_height(p_render_data->environment);
+		scene_state.ubo.fog_depth_curve = environment_get_fog_depth_curve(p_render_data->environment);
```

Comment:
It seems `height` got lost here if `fog_height_density` is greater than 0, then that could explain fog not appearing.